route-rs-runtime/src/link/classify_link.rs

```diff
@@ -167,37 +244,97 @@ mod tests {
         }
     }
 
+    fn run_tokio(runnables: Vec<TokioRunnable>) {
+        tokio::run(lazy(|| {
+            for runnable in runnables {
+                tokio::spawn(runnable);
+            }
+            Ok(())
+        }));
+    }
+
+    #[test]
+    #[should_panic]
+    fn panics_when_built_without_input_streams() {
+        let even_classifier = ClassifyEvenness::new();
+
+        ClassifyLink::new()
+            .branches(10)
+            .classifier(even_classifier)
+            .dispatcher(Box::new(|evenness| if evenness { 0 } else { 1 }))
+            .build_link();
+    }
+
+    #[test]
+    #[should_panic]
+    fn panics_when_built_without_branches() {
+        let packets: Vec<i32> = vec![];
+        let packet_generator: PacketStream<i32> = immediate_stream(packets.clone());
+
+        let even_classifier = ClassifyEvenness::new();
+
+        ClassifyLink::new()
+            .ingressors(vec![Box::new(packet_generator)])
+            .classifier(even_classifier)
+            .dispatcher(Box::new(|evenness| if evenness { 0 } else { 1 }))
+            .build_link();
+    }
+
+    #[test]
+    #[should_panic]
+    fn panics_when_built_without_classifier() {
+        let packets: Vec<i32> = vec![];
+        let packet_generator: PacketStream<i32> = immediate_stream(packets.clone());
+
+        ClassifyLink::<ClassifyEvenness>::new()
+            .ingressors(vec![Box::new(packet_generator)])
+            .branches(10)
+            .dispatcher(Box::new(|evenness| if evenness { 0 } else { 1 }))
+            .build_link();
+    }
+
+    #[test]
+    #[should_panic]
+    fn panics_when_built_without_dispatcher() {
+        let packets: Vec<i32> = vec![];
+        let packet_generator: PacketStream<i32> = immediate_stream(packets.clone());
+
+        let even_classifier = ClassifyEvenness::new();
+
+        ClassifyLink::new()
+            .ingressors(vec![Box::new(packet_generator)])
+            .classifier(even_classifier)
+            .build_link();
+    }
+
     #[test]
     fn even_odd() {
-        let default_channel_size = 10;
         let number_branches = 2;
         let packet_generator = immediate_stream(vec![0, 1, 2, 420, 1337, 3, 4, 5, 6, 7, 8, 9]);
 
         let even_classifier = ClassifyEvenness::new();
 
-        let mut even_link = ClassifyLink::new(
-            packet_generator,
-            even_classifier,
-            Box::new(|evenness| if evenness { 0 } else { 1 }),
-            default_channel_size,
-            number_branches,
-        );
+        let (mut runnables, mut egressors) = ClassifyLink::new()
+            .ingressors(vec![Box::new(packet_generator)])
```

Comment:
`ClassifyLink`'s test examples can also use the simpler `ingressor` builder function as well.

Reply:
done